### PR TITLE
Bug fix 3.7/key tests

### PR DESCRIPTION
**Pull request by dothebart**
backport of https://github.com/arangodb/arangodb/pull/13409

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13779/

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13805/

**Comment by KVS85** [comment]
> http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13805/

Tests blue except unrelated known failure.

**Comment by hkernbach** [comment]
Will start review as soon: https://github.com/arangodb/arangodb/pull/13409 is done.

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13852/

**Review by hkernbach — APPROVED** [review]
LGTM. Ready to merge after Jenkins run. (If Jenkins is green obviously)

**Review by hkernbach — CHANGES_REQUESTED** [review]
@dothebart Ah, one thing: Changelog entry is missing.
otherwise LGTM

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13863/

**Review by hkernbach — APPROVED** [review]

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,8 @@
 v3.7.7 (XXXX-XX-XX)
 -------------------
 
+* refactor dump & restore testsuites; add autoincrement and padded / uuid tests
```
Comment (jsteemann) [inline CHANGELOG:4]:
```suggestion
* Refactored testsuites for dump & restore, and added tests for autoincrement,
  padded and uuid key generators to it.
```

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**